src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs

```diff
@@ -518,32 +544,70 @@ private static List<CentralDirectoryHeaderMetadata> RemoveSignatureAndOrderByOff
 
             return centralDirectoryRecordsList;
         }
-#else
 
-        public static Task<bool> RemoveRepositorySignaturesAsync(
-            Stream input,
-            Stream output,
-            CancellationToken cancellationToken)
+        internal static string GetPackageContentHash(BinaryReader reader)
```

Comment:
should this method receive the hash algorithm so it can make the content hash with any desired algorithm?

Reply:
Will this function be used when creating a signature? or when validating integrity?

Reply:
As I mentioned above, currently we'll only use SHA512 for content hash and it's not configurable.

This func will only be used to get content hash (currently to write down metadata file, but tomorrow for any other purpose as well) but won't be used to create signature or integrity. We have kept supporting dotnet core for signing out of this scope.